### PR TITLE
Code Refactor

### LINKED ISSUE #7: Task `install policy.json in Ubuntu` runs on all distros
I just noticed the following task
```
- name: install policy.json in Ubuntu
      shell: |
              cd /root/src/github.com/kubernetes-incubator/cri-o && \
              cp test/policy.json /etc/containers/policy.json
```

Should this be run on all distros? (remove `in Ubuntu` from name)
Or should this be run only in Ubuntu (add `when: ansible_distribution == "Ubuntu"`)

### PR DESCRIPTION
**Pull request by albertomurillo**
Fix https://github.com/cri-o/cri-o-ansible/issues/7

	Install packages for Fedora/RedHat/CentOS in one task 
	Use update_cache option from apt module
	Insert blank lines between tasks for better readability.
	Shorten comparison of ansible_distribution variable
	Install git on Ubuntu
	Add *.retry to .gitignore
	Check if go is already installed
	Some idempotency fixes

### CODE REVIEW

File: cri-o.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@
         - build-essential
         - curl
         - gawk
+        - git
```
Comment (runcom) [inline cri-o.yml:52]: could you elaborate on why this is needed?

**Comment by runcom** [comment]
@albertomurillo thanks for this! I'm about to merge #5 though so that will cause a rebase for sure :(

File: cri-o.yml (hunk 1 above)
Comment (albertomurillo) [reply cri-o.yml:52]: Well there are tasks to clone git repositories and my Ubuntu installation didn't had git by default.

**Comment by runcom** [comment]
@albertomurillo could you rebase and I'll give this a try asap?

**Comment by runcom** [comment]
the play is stuck at:
```
 ______________________________________
< TASK [Update all packages on Fedora] >
 --------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

skipping: [192.168.122.53]
skipping: [192.168.122.241]

```
it's supposed to `dnf update` in Fedora, but if I ssh in the machine I can `dnf update` myself, meaning no dnf process is already running in there :confused:

**Comment by runcom** [comment]
So I provisioned a 3 node cluster with this playbook in this PR but somehow `kubeadm init` doesn't work anymore, I'm re-trying another cluster with master to understand if this PR is the issue.

**Comment by runcom** [comment]
so it does work with master, I'm retrying one more time with this PR before saying it's an issue with this PR :(

File: cri-o.yml [hunk 2]
```diff
@@ -149,69 +147,75 @@
...
-        insertafter: 'EOF'
-        regexp: '\/net\/bridge\/bridge-nf-call-iptables = 1'
-        state: present
-      ignore_errors: true
-    - name: reload sysctl
-      command: "sysctl -p"
+      sysctl:
+        name: net/bridge/bridge-nf-call-iptables
+        value: 1
+        sysctl_file: /etc/sysctl.conf
```
Comment (runcom) [inline cri-o.yml]: https://docs.ansible.com/ansible/sysctl_module.html can drop this as it's the default

File: cri-o.yml [hunk 3]
```diff
@@ -149,69 +147,75 @@
...
-        line: "net/bridge/bridge-nf-call-iptables = 1"
-        dest: /etc/sysctl.conf
-        insertafter: 'EOF'
-        regexp: '\/net\/bridge\/bridge-nf-call-iptables = 1'
-        state: present
-      ignore_errors: true
-    - name: reload sysctl
-      command: "sysctl -p"
+      sysctl:
+        name: net/bridge/bridge-nf-call-iptables
```
Comment (runcom) [inline cri-o.yml]: this could be `net.bridge.bridge-nf-call-iptables` following https://docs.ansible.com/ansible/sysctl_module.html

**Comment by runcom** [comment]
@albertomurillo have you tested if this is working on Fedora? I couldn't get it work as per our previous IRC chat :(

**Comment by albertomurillo** [comment]
@runcom no but it didnt work either with master... I think is an issue with not restarting crio and kubelet services after `/etc/crio/crio.conf` and `/etc/systemd/system/kubelet.service.d/0-crio.conf` are changed

**Comment by runcom** [comment]
> @runcom no but it didnt work either with master... I think is an issue with not restarting crio and kubelet services after /etc/crio/crio.conf and /etc/systemd/system/kubelet.service.d/0-crio.conf are changed

may be right, actually, we should do a daemon-reload after we install the systemd drop in. WRT crio.conf I think crio isn't started when we edit it, so that's probably ok.

**Comment by albertomurillo** [comment]
I have the change to reload daemons ready since yesterday im testing them right now but my internet is a bit slow :P

**Comment by albertomurillo** [comment]
This is ready now.

File: cri-o.yml [hunk 4]
```diff
@@ -209,9 +212,12 @@
...
     - name: systemd dropin for kubeadm
-      shell: |
-              sh -c 'echo "[Service]
-              Environment=\"KUBELET_EXTRA_ARGS=--enable-cri=true --container-runtime=remote --runtime-request-timeout=15m --image-service-endpoint /var/run/crio.sock --container-runtime-endpoint /var/run/crio.sock\"" > /etc/systemd/system/kubelet.service.d/0-crio.conf'
+      copy:
+        dest: /etc/systemd/system/kubelet.service.d/0-crio.conf
+        content: |
+                  [Service]
+                  Environment="KUBELET_EXTRA_ARGS=--enable-cri=true --container-runtime=remote --runtime-request-timeout=15m --image-service-endpoint /var/run/crio.sock --container-runtime-endpoint /var/run/crio.sock"
+      notify: restart kubelet
```
Comment (runcom) [inline cri-o.yml]: tihs isn't needed, `kubeadm` is taking care of this

Comment (albertomurillo) [reply cri-o.yml]: done

**Comment by runcom** [comment]
I still can't make this to work on Fedora 25 :confused: 

```
[certificates] Generated service account token signing key and public key.
[certificates] Generated front-proxy CA certificate and key.
[certificates] Generated front-proxy client certificate and key.
[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
[apiclient] Created API client, waiting for the control plane to become ready
[apiclient] All control plane components are healthy after 145.514402 seconds
[apiclient] Waiting for at least one node to register
[apiclient] Temporarily unable to list nodes (will retry)
[apiclient] Temporarily unable to list nodes (will retry)
[apiclient] Temporarily unable to list nodes (will retry)
[apiclient] Temporarily unable to list nodes (will retry)
```

**Comment by albertomurillo** [comment]
Im splitting this PR into smaller ones.

**Pull request closed without merging**